archey/entries/uptime.py

```diff
@@ -8,41 +17,147 @@ class Uptime(Entry):
+        uptime_seconds = int(self._get_uptime_delta().total_seconds())
 
-        day, fuptime = divmod(fuptime, 86400)
-        hour, fuptime = divmod(fuptime, 3600)
-        minute = fuptime // 60
+        days, uptime_seconds = divmod(uptime_seconds, 86400)
+        hours, uptime_seconds = divmod(uptime_seconds, 3600)
+        minutes = uptime_seconds // 60
 
         uptime = ''
-        if day:
-            uptime += str(day) + ' day'
-            if day > 1:
+        if days:
+            uptime += str(days) + ' day'
+            if days > 1:
                 uptime += 's'
 
-            if hour or minute:
-                if bool(hour) != bool(minute):
+            if hours or minutes:
+                if bool(hours) != bool(minutes):
                     uptime += ' and '
-
                 else:
                     uptime += ', '
 
-        if hour:
-            uptime += str(hour) + ' hour'
-            if hour > 1:
+        if hours:
+            uptime += str(hours) + ' hour'
+            if hours > 1:
                 uptime += 's'
 
-            if minute:
+            if minutes:
                 uptime += ' and '
 
-        if minute:
-            uptime += str(minute) + ' minute'
-            if minute > 1:
+        if minutes:
+            uptime += str(minutes) + ' minute'
+            if minutes > 1:
                 uptime += 's'
-
-        else:
-            if not day and not hour:
-                uptime = '< 1 minute'
+        elif not days and not hours:
+            uptime = '< 1 minute'
 
         self.value = uptime
+
+    def _get_uptime_delta(self):
+        """
+        Returns a `datetime.timedelta` instance containing the machine uptime.
+        Tries a variety of methods, increasing compatibility for a wide range of systems.
+        """
+        # Try the /proc/uptime file
+        try:
+            return self._proc_file_uptime()
+        except (PermissionError, FileNotFoundError):
+            # Can't read /proc/uptime.
+            # Not GNU/Linux ? Limited permissions ?
+            pass
+
+        # Try the python `time` module clocks
+        try:
+            return self._clock_uptime()
+        except RuntimeError:
+            # Probably Python <3.7, or not *NIX
+            pass
+
+        # FUTURE: Windows support could be added here with the `wmi` or `pywin32` module.
+
+        # Fall back to the `uptime` command.
+        return self._parse_uptime_cmd()
+
+    @staticmethod
+    def _proc_file_uptime():
+        """Tries to get uptime using the `/proc/uptime` file"""
+        with open('/proc/uptime') as f_uptime:
+            return timedelta(
+                seconds=float(f_uptime.read().split()[0])
+            )
+
+    @staticmethod
+    def _clock_uptime():
+        """Tries to get uptime using the clocks from the Python `time` module"""
+        # Try: Linux uptime clock, macOS uptime clock, BSD uptime clock.
+        for clock in ('CLOCK_BOOTTIME', 'CLOCK_UPTIME_RAW', 'CLOCK_UPTIME'):
```

Comment:
`tuple` instead of `list` for constant `for` loops items (the rest of Archey will be migrated as well when possible) in the near future.

Reply:
I'll try and keep that in mind for future commits then 😃